app.rb

```diff
@@ -1,12 +1,78 @@
+require 'aws-sdk-secretsmanager'
 require 'sinatra'
 require 'pry-byebug'
-require "./efiler_service.rb"
+require 'jwt'
+require 'dotenv/load'
+require 'nokogiri'
 
-get '/' do
-  'Hello world!'
+require_relative "efiler_service"
+require_relative "acks"
+require_relative "submissions_status"
+require_relative "extensions/efiler_api_client_auth"
+
+before do
+  content_type 'application/json'
+end
+
+authenticate_jwt
+
+post '/submit' do
+  unless params.has_key?("submission_bundle")
+    status 400
+    return { error_message: "Request missing submission_bundle param" }.to_json
+  end
+
+  submission_filename = params["submission_bundle"]["filename"]
+  result = Dir.mktmpdir do |dir|
+    submission_path = File.join(dir, submission_filename)
+    FileUtils.mv params["submission_bundle"]["tempfile"].path, submission_path
+    EfilerService.run_efiler_command(get_api_client_mef_credentials, "submit", submission_path)
+  end
+
+  doc = Nokogiri::XML(result)
+  if doc.css('SubmissionReceiptList SubmissionReceiptGrp SubmissionId').text.strip == File.basename(submission_filename, ".zip")
```

Comment:
is there ever a possibility that the `doc.css('SubmissionReceiptList SubmissionReceiptGrp SubmissionId')` would return `nil`? I'm guessing maybe not since it's not covered scenario in the existing code

Reply:
The `submission_bundle` nil check is so that we can tell an API client that they sent us a malformed request. The XML checking is looking at the response from MeF (in the same way that vita-min does), and so I expect the `if` check to be false (not raise an exception) in the case that MeF handled our request successfully but did not allow submission. In any case that the response is malformed or some other exceptions is raised, we'll catch that and return a 500.